src/server/plugins/start-and-end/index.js

```diff
@@ -1,48 +1,82 @@
 import _ from 'lodash';
 import cache from '../../config/cache';
 import cookies from '../../config/cookies';
+import practiceLookup from '../../../shared/lib/practice-lookup';
 
-function redirectToStart(request, reply) {
-  reply.redirect('/start');
+function practiceHandler(request, reply) {
+  if (request.state.practice) {
+    return reply.redirect(request.aka('start'));
+  }
+
+  if (request.params.practice) {
+    const practice = practiceLookup.getPractice(request.params.practice);
+
+    if (typeof practice !== 'undefined') {
+      return reply
+        .redirect(request.aka('start'))
+        .state('practice', request.params.practice);
+    }
+  }
+
+  reply
+    .view('practices', {
+      practices: practiceLookup.getPractices(),
+    });
 }
 
 function startHandler(request, reply) {
-  reply.view('index', {
-    message:'Hello, template context!',
-    title: 'hello from nunjucks',
-    time: Date.now()
-  });
+  if (request.state.practice) {
+    reply.view('start');
+  } else {
+    reply.redirect(request.aka('choose'));
+  }
 }
 
 function endHandler(request, reply) {
-  reply.view('end', {});
+  if (request.params.practice) {
+    const practice = practiceLookup.getPractice(request.params.practice);
+
+    if (typeof practice !== 'undefined') {
+      return reply
+        .view('end', {
+          practice,
+        });
+    }
+  }
+
+  return reply.view('end');
 }
 
 exports.register = function(server, options, next) {
   const statelessConfig = {
```

Comment:
can be moved to global server config for cookies as we're not having any pages that are cacheable anymore